File: app/src/main/java/com/karla/learningverbs/viewmodel/LoginViewModel.java
package com.karla.learningverbs.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.karla.learningverbs.repository.userrepository.UserRepository;
import com.karla.learningverbs.utils.firebase.CustomOnCompleteListener;
import com.karla.learningverbs.utils.base.BaseViewModel;
import com.google.firebase.auth.AuthResult;

public class LoginViewModel extends BaseViewModel {
    private MutableLiveData<Boolean> isLogin;
    private UserRepository repository;

    public LoginViewModel() {
        this.repository = UserRepository.getInstance();
        repository = UserRepository.getInstance();
        this.isLogin = new MutableLiveData<>();
    }

    public void doLogin(String email, String password) {
        repository.login(new CustomOnCompleteListener<AuthResult>() {
            @Override
            public void onSuccess(AuthResult task) {
                isLogin.postValue(true);
            }

            @Override
            public void onFailure(Throwable throwable) {
                msgError.postValue(throwable.getMessage());
            }

            @Override
            public void showLoaging() {
                loading.postValue(true);
            }

            @Override
            public void hideLoading() {
                loading.postValue(false);
            }
        }, email, password);

    }

    public LiveData<Boolean> getFireBaseUser() {
        return isLogin;
    }

}

File: app/src/main/java/com/karla/learningverbs/kotlin/viewmodel/LoginViewModel.kt
package com.karla.learningverbs.kotlin.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.google.firebase.auth.AuthResult
import com.karla.learningverbs.kotlin.repository.UserRepository
import com.karla.learningverbs.kotlin.utils.base.BaseViewModel
import com.karla.learningverbs.kotlin.utils.firebase.CustomOnCompleteListener

class LoginViewModel: BaseViewModel() {

    private val isLogin = MutableLiveData<Boolean>()
    private val repository: UserRepository = UserRepository()

    fun doLogin(email: String, password: String) {
        repository.login(object : CustomOnCompleteListener<AuthResult?>() {
            override fun onSuccess(task: AuthResult?) {
                isLogin.postValue(true)
            }

            override fun onFailure(throwable: Throwable?) {
                throwable?.let {
                    msgError.postValue(it.message)

                }
            }

            override fun showLoaging() {
                loading.postValue(true)
            }

            override fun hideLoading() {
                loading.postValue(false)
            }
        }, email, password)
    }

    fun getFireBaseUser(): LiveData<Boolean?> {
        return isLogin
    }

}